packages/playwright-core/src/server/dispatchers/frameDispatcher.ts

```diff
@@ -62,7 +62,7 @@ export class FrameDispatcher extends Dispatcher<Frame, channels.FrameChannel> im
         return;
       const params = { url: event.url, name: event.name, error: event.error ? event.error.message : undefined };
       if (event.newDocument)
-        (params as any).newDocument = { request: RequestDispatcher.fromNullable(this._scope, event.newDocument.request || null) };
+        (params as any).newDocument = { request: RequestDispatcher.fromNullable(scope.parentScope()!, event.newDocument.request || null) };
```

Comment:
This looks like a semantic change.

Reply:
Yes, it was a bug - we were using frame as a parent scope, whereas it should have been the context. Frame has no scope, so its parent is context.